src/app/api/services/guards/auth-guard.service.spec.ts

```diff
@@ -0,0 +1,66 @@
+import {TestBed, fakeAsync, tick} from '@angular/core/testing';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import {ActivatedRouteSnapshot, RouterStateSnapshot} from '@angular/router';
+import { AuthGuardService } from './auth-guard.service';
+import { LoginService } from 'dotcms-js/dotcms-js';
+import { DotRouterService } from './../dot-router-service';
+
+
+@Injectable()
+class MockLoginService {
+    private _isLogin$: Observable<boolean>;
+    get isLogin$() { return this._isLogin$; }
+}
+
+@Injectable()
+class  MockDotRouterService {
+    goToLogin = jasmine.createSpy('goToLogin');
+}
+
+describe('ValidAuthGuardService', () => {
+
+    let authGuardService: AuthGuardService;
+    let dotRouterService: DotRouterService;
+    let loginService: LoginService;
+    let mockRouterStateSnapshot: RouterStateSnapshot;
+    let mockActivatedRouteSnapshot: ActivatedRouteSnapshot;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            providers: [
+                AuthGuardService,
+                { provide: DotRouterService, useClass: MockDotRouterService },
+                { provide: LoginService, useClass: MockLoginService },
+                { provide: RouterStateSnapshot, useValue: mockRouterStateSnapshot},
+                { provide: ActivatedRouteSnapshot, useValue: mockActivatedRouteSnapshot}
+            ]
+        });
+
+        authGuardService = TestBed.get(AuthGuardService);
+        dotRouterService = TestBed.get(DotRouterService);
+        loginService = TestBed.get(LoginService);
+        mockRouterStateSnapshot = jasmine.createSpyObj<RouterStateSnapshot>('RouterStateSnapshot',  ['toString']);
+        mockActivatedRouteSnapshot = jasmine.createSpyObj<ActivatedRouteSnapshot>('ActivatedRouteSnapshot',  ['toString']);
+    });
+
+    it('should allow access to the requested route, User is logged in', fakeAsync(() => {
```

Comment:
You don't need the ```fakeAsync``` here... I tests the resolver this same way and didn't do it.